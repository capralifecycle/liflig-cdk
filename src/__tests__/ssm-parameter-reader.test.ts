import { CfnParameter, ParameterType } from "@aws-cdk/aws-ssm"
import { App, Stack } from "@aws-cdk/core"
import "jest-cdk-snapshot"
import { SsmParameterReader } from "../ssm-parameter-reader"

test("ssm-parameter-reader", () => {
  const app = new App()
  const stack1 = new Stack(app, "Stack1")
  const parameterName = "/my/param"

  new CfnParameter(stack1, "Param", {
    type: ParameterType.STRING,
    name: parameterName,
    value: "test",
  })

  const stack2 = new Stack(app, "Stack2")
  new SsmParameterReader(stack2, "ParamReader", {
    parameterName,
    region: "eu-west-1",
    nonce: "123",
  })

  expect(stack2).toMatchCdkSnapshot()
})
