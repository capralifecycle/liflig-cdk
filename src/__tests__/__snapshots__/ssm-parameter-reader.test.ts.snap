// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ssm-parameter-reader 1`] = `
Object {
  "Mappings": Object {
    "LatestNodeRuntimeMap": Object {
      "af-south-1": Object {
        "value": "nodejs22.x",
      },
      "ap-east-1": Object {
        "value": "nodejs22.x",
      },
      "ap-east-2": Object {
        "value": "nodejs22.x",
      },
      "ap-northeast-1": Object {
        "value": "nodejs22.x",
      },
      "ap-northeast-2": Object {
        "value": "nodejs22.x",
      },
      "ap-northeast-3": Object {
        "value": "nodejs22.x",
      },
      "ap-south-1": Object {
        "value": "nodejs22.x",
      },
      "ap-south-2": Object {
        "value": "nodejs22.x",
      },
      "ap-southeast-1": Object {
        "value": "nodejs22.x",
      },
      "ap-southeast-2": Object {
        "value": "nodejs22.x",
      },
      "ap-southeast-3": Object {
        "value": "nodejs22.x",
      },
      "ap-southeast-4": Object {
        "value": "nodejs22.x",
      },
      "ap-southeast-5": Object {
        "value": "nodejs22.x",
      },
      "ap-southeast-7": Object {
        "value": "nodejs22.x",
      },
      "ca-central-1": Object {
        "value": "nodejs22.x",
      },
      "ca-west-1": Object {
        "value": "nodejs22.x",
      },
      "cn-north-1": Object {
        "value": "nodejs22.x",
      },
      "cn-northwest-1": Object {
        "value": "nodejs22.x",
      },
      "eu-central-1": Object {
        "value": "nodejs22.x",
      },
      "eu-central-2": Object {
        "value": "nodejs22.x",
      },
      "eu-isoe-west-1": Object {
        "value": "nodejs18.x",
      },
      "eu-north-1": Object {
        "value": "nodejs22.x",
      },
      "eu-south-1": Object {
        "value": "nodejs22.x",
      },
      "eu-south-2": Object {
        "value": "nodejs22.x",
      },
      "eu-west-1": Object {
        "value": "nodejs22.x",
      },
      "eu-west-2": Object {
        "value": "nodejs22.x",
      },
      "eu-west-3": Object {
        "value": "nodejs22.x",
      },
      "il-central-1": Object {
        "value": "nodejs22.x",
      },
      "me-central-1": Object {
        "value": "nodejs22.x",
      },
      "me-south-1": Object {
        "value": "nodejs22.x",
      },
      "mx-central-1": Object {
        "value": "nodejs22.x",
      },
      "sa-east-1": Object {
        "value": "nodejs22.x",
      },
      "us-east-1": Object {
        "value": "nodejs22.x",
      },
      "us-east-2": Object {
        "value": "nodejs22.x",
      },
      "us-gov-east-1": Object {
        "value": "nodejs22.x",
      },
      "us-gov-west-1": Object {
        "value": "nodejs22.x",
      },
      "us-iso-east-1": Object {
        "value": "nodejs18.x",
      },
      "us-iso-west-1": Object {
        "value": "nodejs18.x",
      },
      "us-isob-east-1": Object {
        "value": "nodejs18.x",
      },
      "us-isob-west-1": Object {
        "value": "nodejs18.x",
      },
      "us-west-1": Object {
        "value": "nodejs22.x",
      },
      "us-west-2": Object {
        "value": "nodejs22.x",
      },
    },
  },
  "Resources": Object {
    "AWS679f53fac002430cb0da5b7982bd22872D164C4C": Object {
      "DependsOn": Array [
        "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "c099eb4e32cbf1c3da9c45a3b280efe2bed38d27d74aa72702b67d86d1b52354.zip",
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
            "Arn",
          ],
        },
        "Runtime": Object {
          "Fn::FindInMap": Array [
            "LatestNodeRuntimeMap",
            Object {
              "Ref": "AWS::Region",
            },
            "value",
          ],
        },
        "Timeout": 120,
      },
      "Type": "AWS::Lambda::Function",
    },
    "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ParamReaderCustomResourcePolicy6D6224AD": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "ssm:GetParameter",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":ssm:eu-west-1:",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/my/param",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ParamReaderCustomResourcePolicy6D6224AD",
        "Roles": Array [
          Object {
            "Ref": "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "ParamReaderDE9769DB": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "ParamReaderCustomResourcePolicy6D6224AD",
      ],
      "Metadata": Object {
        "Nonce": "123",
      },
      "Properties": Object {
        "Create": "{\\"service\\":\\"SSM\\",\\"action\\":\\"getParameter\\",\\"parameters\\":{\\"Name\\":\\"/my/param\\"},\\"region\\":\\"eu-west-1\\",\\"physicalResourceId\\":{\\"id\\":\\"/my/param\\"}}",
        "InstallLatestAwsSdk": true,
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "AWS679f53fac002430cb0da5b7982bd22872D164C4C",
            "Arn",
          ],
        },
        "Update": "{\\"service\\":\\"SSM\\",\\"action\\":\\"getParameter\\",\\"parameters\\":{\\"Name\\":\\"/my/param\\"},\\"region\\":\\"eu-west-1\\",\\"physicalResourceId\\":{\\"id\\":\\"/my/param\\"}}",
      },
      "Type": "Custom::AWS",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
